watchtower-plugin/src/net/http.rs

```diff
@@ -154,6 +154,74 @@ pub async fn send_appointment(
     }
 }
 
+/// Handles a ping request to the tower.
+pub async fn ping(tower_net_addr: &NetAddr) -> Result<(), RequestError> {
+    let response = get_request(tower_net_addr, Endpoint::Ping, &None).await?;
+
+    if response.status().is_success() {
+        log::debug!("Ping successful!");
+        Ok(())
+    } else {
+        Err(RequestError::Unexpected(format!(
+            "Ping failed with status code: {}\nError message: {}",
+            response.status(),
+            response
+                .text()
+                .await
+                .map_err(|err| RequestError::Unexpected(format!(
+                    "Failed to get error message: {}",
+                    err
+                )))?
+        )))
+    }
+}
+
+/// Generic function to get different types of requests from the tower.
+pub async fn get_request(
```

Comment:
This is almost a copy, line by line, of the post_request method.

The only differences are:

- the `.get()` method is called instead of `post()`
- `json(&data)` is not called

This calls for a generalization of the `post_request` method to be called something like `request` and conditionally being called with a flag indicating whether `post` or `get` wants to be called:

we could have something on the lines of:

```
pub async fn post_request<S: Serialize>(
    tower_net_addr: &NetAddr,
    endpoint: Endpoint,
    data: S,
    proxy: &Option<ProxyInfo>,
) -> Result<Response, RequestError>;
pub async fn get_request(
    tower_net_addr: &NetAddr,
    endpoint: Endpoint,
    proxy: &Option<ProxyInfo>,
) -> Result<Response, RequestError>;
pub async fn request(
    tower_net_addr: &NetAddr,
    endpoint: Endpoint,
    proxy: &Option<ProxyInfo>,
    method: Method, 
) -> Result<Response, RequestError>;
```

And let method be: https://docs.rs/reqwest/latest/reqwest/struct.Method.html#

Therefore, `post_request` will call `request` with `method=Method::POST` and `get_request` will call `request` with  `method=Method:GET`

Reply:
Notice tests will need to be created/updated too

Reply:
Oh yess! I will make `request` and configure `post_request` and `get_request` as suggested!

Reply:
https://docs.rs/reqwest/latest/reqwest/struct.Method.html# is unavailable. could you please point again?